### PR TITLE
Implement Ed25519 and Ed448 in the default provider

### PR DESCRIPTION
**Pull request by mattcaswell**
Implements Ed25519 and Ed448 in the default provider. In order to do this we have to add support for the "one shot" EVP_DigestSign() and EVP_DigestVerify() functions for providers. These two algorithms only support the one shot method.

Note: there is no support for serializers yet. That will come in a subsequent PR.

### CODE REVIEW

**Comment by mattcaswell** [comment]
Ping?

File: providers/implementations/signature/eddsa.c [hunk 1]
```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (slontis) [inline providers/implementations/signature/eddsa.c]: 2020

**Comment by mattcaswell** [comment]
Nit fixed. Fixup commit pushed. Please take another look.

**Comment by mattcaswell** [comment]
Pushed. Thanks!

**Comment by kroeckx** [comment]
Are there plans to add support for this in the FIPS provider?

I hope that FIPS 186-5 and SP 800-186 will be published soon.

**Comment by mattcaswell** [comment]
> Are there plans to add support for this in the FIPS provider?

They're not currently on the list for inclusion. See Appendix 3 here:

https://www.openssl.org/docs/OpenSSL300Design.html

**Comment by mattcaswell** [comment]
> They're not currently on the list for inclusion. See Appendix 3 here:

Although apparently #11371 is adding them to the FIPS provider as "non-FIPS algorithms"

**Pull request closed without merging**